### PR TITLE
Fix string recognition of ro channel map because of representation sa…

### LINKED ISSUE #17: All parameters are saved using repr instead of str in hdf5 after recent updates
This is undesired as it breaks much of existing analysis code.

The fix for now is to revert `measurement_control.py` line 1161:
```python
for (p_name, p) in parameter_list:
    try:
        val = repr(p['value'])
    except KeyError:
        val = ''
    instrument_grp.attrs[p_name] = val
```
to
```python
for (p_name, p) in parameter_list:
    try:
        val = str(p['value'])
    except KeyError:
        val = ''
    instrument_grp.attrs[p_name] = val
```

But this might break saving of list parameters (e.g. readout freq. for multichromatic pulse)

### PR DESCRIPTION
**Pull request by nathlacroix**
Fixes #17
The readout channel map was wrongly acquired due to the string representation of the string. 

The pycqed code has been revised and no other instances have been found with similar issues, as all other cases look for parameter values (i.e. float) where repr(value) == str(value). If accessing a string value from the HDF file, one should apply eval on it.

### CODE REVIEW

**Comment by antsr** [comment]
But `acq_weights_type` is not the only string parameter in the hdf5 files. We have all the pulse types and channel names, for example.

**Comment by stephlazar** [comment]
~@antsr that is true, but I think what Nathan did is to fix all instances in PycQED where string~ ~parameters are extracted from HDF5 file, by adding eval. In the future, we should always eval~ ~whenever we extract.~

@antsr raises a very good point. I didn't notice this pull request only changes one line in one file. Someone must correct all placed in the PycQED repository where we extract string parameters from HDF5 file (there are several other places where this is done). I can take care of this.

**Comment by nathlacroix** [comment]
All the instances I have found (approx. 300) are either outdated functions in the analysis toolbox or they look for parameters which are numbers, hence eval(value) == value . But maybe I overlooked some of them. We can check that together @stephlazar

**Comment by stephlazar** [comment]
@antsr @nathlacroix I made more changes. Please review.

File: pycqed/analysis/analysis_toolbox.py [hunk 1]
```diff
@@ -344,10 +350,11 @@ def get_qb_channel_map_from_hdf(qb_names, file_path, value_names, h5mode='r+'):
         ro_type = 'w'
 
     for qbn in qb_names:
-        qbchs = [str(instr_settings[qbn].attrs['acq_I_channel'])]
-        ro_acq_weight_type = instr_settings[qbn].attrs['acq_weights_type']
+        qbchs = [str(eval(instr_settings[qbn].attrs['acq_I_channel']))]
```
Comment (antsr) [inline pycqed/analysis/analysis_toolbox.py:353]: `str` should not be needed anymore

File: pycqed/analysis/analysis_toolbox.py [hunk 2]
```diff
@@ -317,7 +322,8 @@ def get_param_value_from_file(file_path, instr_name, param_name, h5mode='r+'):
     instr_settings = data_file['Instrument settings']
     if instr_name in list(instr_settings.keys()):
         if param_name in list(instr_settings[instr_name].attrs):
-            param_val = float(instr_settings[instr_name].attrs[param_name])
+            param_val = float(eval(instr_settings[instr_name].attrs[
```
Comment (antsr) [inline pycqed/analysis/analysis_toolbox.py]: `float` should not be needed anymore with eval

File: pycqed/analysis/measurement_analysis.py [hunk 3]
```diff
@@ -11,58 +10,46 @@
...
 import importlib
-import math
 from time import time
 
 try:
     import pygsti
 except ImportError as e:
     if str(e).find('pygsti') >= 0:
-        logging.warning('Could not import pygsti')
+        logger.warning('Could not import pygsti')
```
Comment (antsr) [inline pycqed/analysis/measurement_analysis.py]: Not a critical issue, but just to raise awareness, that the logger object is usually (in other modules of PycQED) called `log`. I know it was a problem already before.

File: pycqed/instrument_drivers/meta_instrument/FluxDecoupler.py [hunk 4]
```diff
@@ -1,136 +0,0 @@
-import numpy as np
```
Comment (antsr) [inline pycqed/instrument_drivers/meta_instrument/FluxDecoupler.py]: This class should still be working. I wrote it at some point to store and use the flux decoupling matrix. I would not delete it.

**Review by antsr — CHANGES_REQUESTED** [review]
thanks for the changes, @stephlazar 
the `FluxDecoupler` class should not be deleted, it should still work and might be useful.
Also, I think that when we do an `eval`, we shouldn't have to do any further casting, e.g. `str(eval(...` and `float(eval(...` and `int(eval(...` should be replaced with just `eval(...`

**Review by nathlacroix — COMMENTED** [review]
@stephlazar thanks for the changes and the clean up !
I will address the comments of @antsr asap (this weekend or next week)

**Comment by stephlazar** [comment]
@nathlacroix I already did :)

**Review by antsr — APPROVED** [review]

**Review by stephlazar — DISMISSED** [review]
Now it looks ok

**Comment by stephlazar** [comment]
I have now tested a few standard analyses that we run and fixed a small bug.
Analyses tested: Rabi with SSB weights, Rabi with classifier and active reset, Ramsey (SSB), QScale (SSB), T1 (SSB), Flux pulse scope, Chevron. 

The pull request can be merged from my point of view.
@antsr @nathlacroix

**Pull request merged**